### PR TITLE
Demo for the topology generator

### PR DESCRIPTION
**Pull request by leandroberetta**
A topology generator where with some parameters, a topology can be generated:

gen1

The demo does not interact with cluster, only generate a list of manifests for the user to create/apply. I guess this is simpler and less invasive.

An example of a topology can be:

kiali-gen-n9

### CODE REVIEW

File: topology-generator/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+This demo has been tested using [Minikube](https://istio.io/latest/docs/setup/platform-setup/minikube/) and [Istio 1.12 Demo Profile](https://istio.io/latest/docs/setup/install/istioctl/#install-a-different-profile).
+
+## Application Install
+
+Create the application resources:
+
+```bash
+kubectl create ns topology-generator
+kubectl apply -n topology-generator https://raw.githubusercontent.com/kiali/demos/master/topology-generator/generator.yaml
```
Comment (jshaughn) [inline topology-generator/README.md:20]:
```suggestion
kubectl apply -n topology-generator -f https://raw.githubusercontent.com/kiali/demos/master/topology-generator/generator.yaml 
```

**Review by jshaughn — COMMENTED** [review]
Worked great for me.  Tested with KinD and Istio 1.11.  One trivial fix requested.  Where is the actual generator/ui code?

**Comment by jshaughn** [comment]
>  but if you want I can move the code here

I just expected to see it as part of the demo.  If you prefer to keep it in your private repo it's fine by me, although it seems disconnected.  You will still own the copyright if it's in the Kiali repo :-)

**Comment by lucasponce** [comment]
> You will still own the copyright if it's in the Kiali repo :-)

+1

The copyright is always for the author of the commit, so no problem in that sense, I'd say that I like to see the code, to invite community or other folks to modify and update the examples.

If there are concerns about "hey, it's a pet project" or similar, the "demos" repo work as a playground and it's pretty flexible to add experiments.

The peer reviews on these repo takes that into consideration and anyone can propose any technology for a demo.

Also, the images can be moved to the Kiali account as well if needed (just to let others or the CI to update/publish new changes in the future)

**Comment by leandroberetta** [comment]
@lucasponce @jshaughn there is no concern about the copyright at all, it's a pleasure to contribute code and store it here, also it would be awesome if people contribute to this demo!.

I believed that it was cleaner and simpler to have less code here, that is isually not used in the demo as we only require the built images but I agree with you, so the code is now here.

Also, I can publish the images to Kiali's Quay but I guess I need permissions to do that.

**Comment by lucasponce** [comment]
> Also, I can publish the images to Kiali's Quay but I guess I need permissions to do that.

Perhaps @jmazzitelli can help you to add you in the kiali organization of quay repo IIRC.

**Review by lucasponce — APPROVED** [review]
Not relevant comments from my side.
The demo is nice, is being used by some folks and the code is attached, so I'm +1 to consolidate and people can iterate on this.

No need to publish on quay on this step, that can be a separated task from this PR.

**Comment by lucasponce** [comment]
Merging to consolidate to make it available, don't hesitate to add any needed update on it.
I understand that demos may have iterations on the go.

**Pull request merged**